edit portfolio.html

```diff
@@ -0,0 +1,149 @@
+              id="description"
+              rows="4"
+              placeholder="Description"
+              required
+            ></textarea>
+            <button id="saveButton">Save</button>
+          </div>
+        </div>
+      </div>
+
+      <div class="side-bar">
+        <div class="avatar">
+          <img src="assets/avatar.jpeg" />
+          <h2>Mishal Firdous</h2>
+          <ul class="a">
+            <li>
+              <label for="file">Wordpress, Wix, Shopify </label>
+              <progress id="file" value="90" max="100">90%</progress>
+            </li>
+
+            <li>
+              <label for="file">HTML5 ,CSS3, Bootstrap</label>
+              <progress id="file" value="70" max="100">32%</progress>
+            </li>
+
+            <li>
+              <label for="file">Adobe Photo Shop </label>
+              <progress id="file" value="30" max="100">32%</progress>
+            </li>
+          </ul>
+        </div>
+      </div>
+    </div>
+    <footer class="linkedIn">
+      <h3>
+        <a href="https://www.linkedin.com/in/mishal-firdous-b99a0a131"
+          >LinkedIn</a
+        >
+      </h3>
+    </footer>
+    <script src="./displayLocalStorageData.js"></script>
+    <script>
+      document.addEventListener("DOMContentLoaded", function () {
+        const addButton = document.getElementById("addButton");
+        const experienceForm = document.getElementById("experienceForm");
+        const saveButton = document.getElementById("saveButton");
+        const experienceList = document.getElementById("experienceList");
+        //store data in localStorage
+        const dataValue =
+          JSON.parse(localStorage.getItem("experiencedata")) || [];
+
+        addButton.addEventListener("click", function () {
+          event.preventDefault();
+          experienceForm.style.display = "block";
+        });
+
+        saveButton.addEventListener("click", function () {
+          event.preventDefault();
+          const companyName = document.getElementById("companyName").value;
+          const startDate = document.getElementById("startDate").value;
+          const endDate = document.getElementById("endDate").value;
+          const description = document.getElementById("description").value;
+          //object creation
+          const experienceData = {
+            companyName,
+            startDate,
+            endDate,
+            description,
+          };
+
+          dataValue.push(JSON.stringify(experienceData));
+          localStorage.setItem("experiencedata", JSON.stringify(dataValue));
+
+          displayLocalStorageData();
+
+          document.getElementById("companyName").value = "";
+          document.getElementById("startDate").value = "";
+          document.getElementById("endDate").value = "";
+          document.getElementById("description").value = "";
+
+          experienceForm.style.display = "none";
+        });
+        displayLocalStorageData();
+      });
```

Comment:
you are mixing things together. for one functionality you have a separate .js file (fetching and displaying experience from local storage) which is the suggested and more cleaner approach but here you are implementing JS functionalities inside script tags in your markup file. Keep things consistent and cleaner and move this logic out of script tag to a javascript file.